### PR TITLE
TST: Enable NPY_RELAXED_STRIDES_DEBUG environment variable.

### PR DESCRIPTION
**Pull request by charris**
Setting NPY_RELAXED_STRIDES_DEBUG=1 in the enviroment when relaxed
stride checking is enabled will cause numpy to be compiled with affected
strides set to bogus values in order to help smoke out incorrect usage
of strides in downstream projects.

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
Needs a release note, but otherwise looks good to me.

**Comment by charris** [comment]
Heh, fair catch ;) Fixed.

File: doc/release/1.13.0-notes.rst
Comment (eric-wieser) [inline doc/release/1.13.0-notes.rst]: There are now 3 or 4 PRs that make this change...

File: doc/release/1.13.0-notes.rst
Comment (eric-wieser) [inline doc/release/1.13.0-notes.rst]: Might be worth explaining what a typical "smoke out" looks like, and specifying that "bogus" means "extremely large".

Could do with a comma after "bogus values", too.

**Comment by charris** [comment]
@matthew-brett If you are building daily wheels for testing purposed you may want to put
`NPY_RELAXED_STRIDES_DEBUG=1` in the build environment.

File: doc/release/1.13.0-notes.rst
Comment (charris) [inline doc/release/1.13.0-notes.rst]: To be sure. IIRC correctly, the usual is an IndexError but I am not sure.

File: doc/release/1.13.0-notes.rst
Comment (charris) [inline doc/release/1.13.0-notes.rst]: Well, one of them has to go first ;) Easy fix on line if homu complains, but now I'm actually curious if it will cause a merge problem.

File: doc/release/1.13.0-notes.rst
Comment (eric-wieser) [inline doc/release/1.13.0-notes.rst]: Still would like a comma after `npy_intp`, but not too important

File: doc/release/1.13.0-notes.rst
Comment (eric-wieser) [inline doc/release/1.13.0-notes.rst]: Yep, and this one looks the least controversial

**Comment by eric-wieser** [comment]
Probably good to wait for a second opinion from someone who's used this before, but I'd be ok with this being merged as is.

File: doc/release/1.13.0-notes.rst
Comment (charris) [inline doc/release/1.13.0-notes.rst]: When a subordinate clause follows an independent clause, a comma is not needed.

I used a comma in the sentence above because the subordinate clause comes first.

Two examples, two cases. Two sentence fragments ;)

Comment (eric-wieser) [inline doc/release/1.13.0-notes.rst]: Oh, I agree it's not necessary (thanks for the formal explanation though) - but am I right in saying it would be valid?

**Comment by charris** [comment]
It was discussed on the list. AFAIK, I'm the only one who has made use of the two options and carrying along a patch to apply to releases was getting old. If Mathew enables the option for his nightly wheel build that is used for testing, there will not be much change.

**Pull request merged**
